File: beakjoon/src/main/java/Problem1874.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;
import java.util.Stack;

public class Problem1874 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());

        Stack<Integer> stack = new Stack<>();
        int count = 1;
        List<String> result = new ArrayList<>();

        for (int i = 0; i < n; i++) {
            int data = Integer.parseInt(br.readLine());
            while (data >= count) {
                stack.add(count++);
                result.add("+");
            }
            if (stack.peek() == data) {
                stack.pop();
                result.add("-");
            } else {
                System.out.print("NO");
                return;
            }
        }
        System.out.print(String.join("\n", result));
    }
}

File: beakjoon/src/main/nodejs/Problem1874.js
let lines = [];

require('readline')
    .createInterface(process.stdin, process.stdout)
    .on('line', function (line) {
        lines.push(line);
    })
    .on('close', function () {
        const n = parseInt(lines[0], 10);
        const arr = lines.slice(1).map(num => parseInt(num, 10));
        console.log(solve(n, arr));
    });

function solve(n, arr) {
    let result = [];
    let stack = [];
    let cnt = 1;
    for (let i = 0; i < n; i++) {
        const num = arr[i];
        while (cnt <= num) {
            stack.push(cnt++);
            result.push('+');
        }
        if (stack.length > 0 && stack[stack.length - 1] === num) {
            stack.pop();
            result.push('-');
        }
    }

    if(stack.length === 0) {
        return result.join('\n');
    } else {
        return 'NO';
    }
}

module.exports = solve;
